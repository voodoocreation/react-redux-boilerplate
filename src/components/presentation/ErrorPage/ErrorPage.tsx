import Head from "next/head";
import * as React from "react";
import { InjectedIntl, injectIntl } from "react-intl";

interface IProps {
  intl: InjectedIntl;
  message?: string;
  status: number;
}

class ErrorPage extends React.Component<IProps> {
  public static defaultProps = {
    status: 500
  };

  public render() {
    const { formatMessage } = this.props.intl;

    return (
      <React.Fragment>
        <Head>
          <title>
            {`${this.getTitle()} Â· ${formatMessage({
              id: "BRAND_NAME"
            })}`}
          </title>
        </Head>

        <h1>{this.getTitle()}</h1>

        <p>{this.getMessage()}</p>
      </React.Fragment>
    );
  }

  private getTitle() {
    const { status } = this.props;
    const { formatMessage } = this.props.intl;

    switch (status) {
      default:
        return formatMessage({ id: "ERROR_TITLE" });

      case 404:
        return formatMessage({ id: "ERROR_404_TITLE" });
    }
  }

  private getMessage() {
    const { message, status } = this.props;
    const { formatMessage } = this.props.intl;

    switch (status) {
      default:
        return message ? message : formatMessage({ id: "ERROR_MESSAGE" });

      case 404:
        return formatMessage({ id: "ERROR_404_MESSAGE" });
    }
  }
}

export default injectIntl<any>(ErrorPage);
