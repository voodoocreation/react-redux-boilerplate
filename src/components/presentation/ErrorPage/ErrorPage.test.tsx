import { render } from "enzyme";
import * as React from "react";

import ErrorPage from "./ErrorPage";

const setup = (fn: any, fromTestProps?: any) => {
  const props = {
    ...fromTestProps
  };

  return {
    actual: fn(<ErrorPage {...props} />),
    props
  };
};

describe("<ErrorPage />", () => {
  it("renders 500 correctly", () => {
    const { actual } = setup(render);
    expect(actual).toMatchSnapshot();
  });

  it("renders 404 correctly", () => {
    const { actual } = setup(render, {
      message: "Not found",
      status: 404
    });
    expect(actual).toMatchSnapshot();
  });
});
