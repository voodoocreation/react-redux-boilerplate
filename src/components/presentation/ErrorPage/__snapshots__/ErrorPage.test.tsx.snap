// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorPage /> renders 404 correctly 1`] = `
Array [
  <h1>
    Page not found
  </h1>,
  <p>
    This page does not exist.
  </p>,
]
`;

exports[`<ErrorPage /> renders 500 correctly 1`] = `
Array [
  <h1>
    An error has occurred
  </h1>,
  <p>
    An error has occurred, please try again.
  </p>,
]
`;
